### PR TITLE
[istio]: Bump istio version to 1.21.6 and Kiali to 1.81

### PR DESCRIPTION
**Pull request by skurbatov**
## Description
Add istio version 1.21.6 ([Changelog](https://istio.io/latest/news/releases/1.21.x/)).
Bump kiali version from 1.67 to 1.81 ([Changelog](https://kiali.io/news/release-notes/#1810))
Istio 1.16 is deprecated.
Add requirements for Deckhouse updates (Required version istio 1.19 for Deckhouse upgrade to next release).

## Why do we need it, and what problem does it solve?
Istio 1.21.6 and Kiali 1.81 releases contain bug fixes to improve robustness.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [x] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: isio
type: feature
summary: Bump Istio version to `1.21.6`. Bump Kiali version to `1.81`.
impact: In environments where legacy versions of istio are used, the D8 update will be blocked, and the `D8IstioDeprecatedIstioVersionInstalled` alert will be fired.
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @skurbatov. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @skurbatov. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @skurbatov. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @skurbatov. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @skurbatov. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @skurbatov. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @skurbatov. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @skurbatov. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @skurbatov. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @skurbatov. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Review by name212 — APPROVED** [review]

**Review by sprait — APPROVED** [review]

**Review by z9r5 — APPROVED** [review]

**Review by tarabrind — APPROVED** [review]

File: modules/110-istio/images/operator-v1x19x7/patches/apply_go.patch [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
++++ b/operator/pkg/helmreconciler/apply.go
+@@ -235,7 +235,12 @@ func (h *HelmReconciler) serverSideApply(obj *unstructured.Unstructured) error {
+ 	scope.Infof("using server side apply to update obj: %v", objectStr)
+ 	opts := []client.PatchOption{client.ForceOwnership, client.FieldOwner(fieldOwnerOperator)}
+ 	if err := h.client.Patch(context.TODO(), obj, client.Apply, opts...); err != nil {
+-		return fmt.Errorf("failed to update resource with server-side apply for obj %v: %v", objectStr, err)
++		if !(strings.Contains(err.Error(), `cannot patch resource "rolebindings"`) ||
++			strings.Contains(err.Error(), `cannot patch resource "roles"`) ||
++			strings.Contains(err.Error(), `cannot patch resource "clusterroles"`) ||
++			strings.Contains(err.Error(), `cannot patch resource "clusterrolebindings"`)) {
```
Comment (Beastlex) [inline modules/110-istio/images/operator-v1x19x7/patches/apply_go.patch:13]:
```suggestion
+		if errors.IsForbidden(err) {
```

Comment (Beastlex) [reply modules/110-istio/images/operator-v1x19x7/patches/apply_go.patch:13]: We have same error type for all these resource: "is forbidden..."

**Review by Beastlex — CHANGES_REQUESTED** [review]

File: modules/110-istio/images/operator-v1x19x7/patches/apply_go.patch (hunk 1 above)
Comment (Beastlex) [reply modules/110-istio/images/operator-v1x19x7/patches/apply_go.patch:13]: `!errors.IsForbidden(...)`

Comment (skurbatov) [reply modules/110-istio/images/operator-v1x19x7/patches/apply_go.patch]: Fixed

**Review by Beastlex — APPROVED** [review]

File: modules/110-istio/images/cni-v1x21x6/Dockerfile [hunk 2]
```diff
@@ -1,8 +1,8 @@
-# Based on https://github.com/istio/istio/blob/1.16.2/cni/deployments/kubernetes/Dockerfile.install-cni
-ARG BASE_DISTROLESS
-FROM docker.io/istio/install-cni:1.16.2@sha256:14255494cf4c78740913c702aa0768fad4b433e82d10cdfc264ef5f30cf041ec as artifact
+# Based on https://github.com/istio/istio/blob/1.21.6/cni/deployments/kubernetes/Dockerfile.install-cni
+ARG BASE_ALT
```
Comment (AndreyPavlovFlant) [inline modules/110-istio/images/cni-v1x21x6/Dockerfile:2]: Why did you change `BASE_DISTROLESS` to `BASE_ALT` here and elsewhere?
`BASE_DISTROLESS' is the preferred option.

File: modules/110-istio/images/operator-v1x19x7/werf.inc.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    - '**/*.patch'
+mount:
+- fromPath: ~/go-pkg-cache
+  to: /go/pkg
+shell:
+  install:
+  - git clone --depth 1 --branch {{ $istioOperatorVersion }} https://github.com/istio/istio.git /tmp/istio/
+  - cd /tmp/istio/
+  - find /patches -name '*.patch' | xargs git apply --verbose
+  - echo 1.19.7 > version
```
Comment (AndreyPavlovFlant) [inline modules/110-istio/images/operator-v1x19x7/werf.inc.yaml:38]:
```suggestion
  - echo {{ $istioOperatorVersion }} > version
```

File: modules/110-istio/images/operator-v1x19x7/werf.inc.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  after: setup
+- artifact: {{ .ModuleName }}/{{ .ImageName }}-src-artifact
+  add: /var/lib/istio/manifests/
+  to: /var/lib/istio/manifests/
+  after: setup
+docker:
+  USER: "1337:1337"
+  ENTRYPOINT: ["/usr/local/bin/operator"]
+---
+artifact: {{ .ModuleName }}/{{ .ImageName }}-build-artifact
```
Comment (AndreyPavlovFlant) [inline modules/110-istio/images/operator-v1x19x7/werf.inc.yaml:20]:
```suggestion
image: {{ .ModuleName }}/{{ .ImageName }}-build-artifact
final: false
```

File: modules/110-istio/images/operator-v1x19x7/werf.inc.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  install:
+  - git clone --depth 1 --branch {{ $istioOperatorVersion }} https://github.com/istio/istio.git /tmp/istio/
+  - cd /tmp/istio/
+  - find /patches -name '*.patch' | xargs git apply --verbose
+  - echo 1.19.7 > version
+  - export GOPROXY={{ $.GOPROXY }}
+  - GOOS=linux GOARCH=amd64 LDFLAGS='-extldflags -static -s -w' common/scripts/gobuild.sh /tmp/istio/out/ -tags=agent,disable_pgv /tmp/istio/operator/cmd/operator/
+  - chown 0700 /tmp/istio/out/operator
+---
+artifact: {{ .ModuleName }}/{{ .ImageName }}-src-artifact
```
Comment (AndreyPavlovFlant) [inline modules/110-istio/images/operator-v1x19x7/werf.inc.yaml:43]:
```suggestion
image: {{ .ModuleName }}/{{ .ImageName }}-binary-artifact
final: false
```

File: modules/110-istio/images/operator-v1x19x7/werf.inc.yaml [hunk 6]
```diff
@@ -0,0 +1,44 @@
+---
+{{- $istioOperatorVersion := "1.19.7" }}
+---
+# Based on https://github.com/istio/istio/blob/1.19.7/operator/docker/Dockerfile.operator
+image: {{ .ModuleName }}/{{ .ImageName }}
+from: {{ $.Images.BASE_ALT }}
+import:
+- artifact: {{ .ModuleName }}/{{ .ImageName }}-build-artifact
```
Comment (AndreyPavlovFlant) [inline modules/110-istio/images/operator-v1x19x7/werf.inc.yaml:8]:
```suggestion
- image: {{ .ModuleName }}/{{ .ImageName }}-build-artifact
```

File: modules/110-istio/images/operator-v1x19x7/werf.inc.yaml [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  add: /tmp/istio/out/operator
+  to: /usr/local/bin/
+  after: setup
+- artifact: {{ .ModuleName }}/{{ .ImageName }}-src-artifact
```
Comment (AndreyPavlovFlant) [inline modules/110-istio/images/operator-v1x19x7/werf.inc.yaml:12]:
```suggestion
- image: {{ .ModuleName }}/{{ .ImageName }}-binary-artifact
```

File: modules/110-istio/images/operator-v1x21x6/werf.inc.yaml [hunk 8]
```diff
@@ -0,0 +1,44 @@
+---
```
Comment (AndreyPavlovFlant) [inline modules/110-istio/images/operator-v1x21x6/werf.inc.yaml:1]: Make the same edits as in images/operator-v1x19x7/werf.inc.yaml

**Review by AndreyPavlovFlant — DISMISSED** [review]

File: modules/110-istio/images/operator-v1x21x6/werf.inc.yaml (hunk 8 above)
Comment (skurbatov) [reply modules/110-istio/images/operator-v1x21x6/werf.inc.yaml:1]: Done

File: modules/110-istio/images/cni-v1x21x6/Dockerfile (hunk 2 above)
Comment (skurbatov) [reply modules/110-istio/images/cni-v1x21x6/Dockerfile]: Return Distroless in operators werf manifests. In other image need research for migrate to distroless.

File: modules/040-control-plane-manager/hooks/audit_policy_test.go [hunk 9]
```diff
@@ -59,8 +59,8 @@ metadata:
 data:
   basicAuditPolicy: |
     serviceAccounts:
+    - system:serviceaccount:d8-istio:istiod-v1x21x6
     - system:serviceaccount:d8-istio:istiod-v1x19x7
-    - system:serviceaccount:d8-istio:istiod-v1x16x9
```
[the remark this answers is not in the document]
Comment (skurbatov) [reply modules/040-control-plane-manager/hooks/audit_policy_test.go:63]: Fixed

File: modules/110-istio/docs/README.md [hunk 10]
```diff
@@ -9,8 +9,8 @@ webIfaces:
 
 | Istio version | [K8S versions supported by Istio](https://istio.io/latest/docs/releases/supported-releases/#support-status-of-istio-releases) |          Status in D8          |
 |:-------------:|:-----------------------------------------------------------------------------------------------------------------------------:|:------------------------------:|
-|     1.16      |                                                1.22<sup>*</sup>, 1.23<sup>*</sup>, 1.24<sup>*</sup>, 1.25                                                | Deprecated and will be deleted |
-|     1.19      |                                                    1.25, 1.26, 1.27, 1.28                                                     |           Supported            |
+|     1.21      |                                                1.26, 1.27, 1.28, 1.29                                                | Supported |
```
[the remark this answers is not in the document]
Comment (apolovov) [reply modules/110-istio/docs/README.md:12]: Just add them.

File: modules/110-istio/docs/README_RU.md [hunk 11]
```diff
@@ -9,8 +9,8 @@ webIfaces:
 
 | Версия Istio | Версии K8S, поддерживаемые Istio | Статус в текущем релизе D8 |
 |:------------:|:------------------------------------------------------------------------------------------------------------------------------:|:--------------------------:|
-|     1.16     |                                           1.22<sup>*</sup>, 1.23<sup>*</sup>, 1.24<sup>*</sup>, 1.25<sup>*</sup>                                            |  Устарела и будет удалена  |
-|     1.19     |                                                     1.25<sup>*</sup>, 1.26, 1.27, 1.28, 1.29, 1.30                                                     |       Поддерживается       |
+|     1.21     |                                           1.26, 1.27, 1.28, 1.29, 1.30                                            |  Поддерживается  |
```
[the remark this answers is not in the document]
Comment (apolovov) [reply modules/110-istio/docs/README_RU.md:12]: Just add it.

File: modules/110-istio/docs/README_RU.md [hunk 12]
```diff
@@ -9,8 +9,8 @@ webIfaces:
...
+|     1.19     |                                                     1.25, 1.26, 1.27, 1.28, 1.28, 1.29, 1.30                                                     |       Устарела и будет удалена       |
```
Comment (apolovov) [inline modules/110-istio/docs/README_RU.md:13]:
```suggestion
|     1.19     |                                                     1.25<sup>*</sup>, 1.26, 1.27, 1.28, 1.28, 1.29, 1.30                                                     |       Устарела и будет удалена       |
```

File: modules/110-istio/docs/README.md [hunk 13]
```diff
@@ -9,8 +9,8 @@ webIfaces:
...
+|     1.19      |                                                    1.25, 1.26, 1.27, 1.28                                                     |           Deprecated and will be deleted            |
```
Comment (apolovov) [inline modules/110-istio/docs/README.md:13]:
```suggestion
|     1.19      |                                                    1.25<sup>*</sup>, 1.26, 1.27, 1.28                                                     |           Deprecated and will be deleted            |
```

File: modules/110-istio/docs/README.md (hunk 10 above)
Comment (apolovov) [inline modules/110-istio/docs/README.md:12]:
```suggestion
|     1.21      |                                                1.26, 1.27, 1.28, 1.29, 1.30, 1.31                                                | Supported |
```

File: modules/110-istio/docs/README_RU.md (hunk 11 above)
Comment (apolovov) [inline modules/110-istio/docs/README_RU.md:12]:
```suggestion
|     1.21     |                                           1.26, 1.27, 1.28, 1.29, 1.30, 1.31                                          |  Поддерживается  |
```

**Review by apolovov — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [11164](https://github.com/deckhouse/deckhouse/pull/11164) to the branch [release-1.67](https://github.com/deckhouse/deckhouse/tree/release-1.67) successful!
